src/GraphTopology.cpp

```diff
@@ -444,4 +481,54 @@ GraphTopology::GraphTopology(const mfem::SparseMatrix& face_edge,
     face_Agg_.Swap(tmp);
 }
 
+GraphTopology::GraphTopology(GraphTopology&& graph_topology) noexcept
+    : edge_d_td_(graph_topology.edge_d_td_)
+{
+    face_d_td_ = std::move(graph_topology.face_d_td_);
+    face_d_td_d_ = std::move(graph_topology.face_d_td_d_);
+
+    Agg_edge_.Swap(graph_topology.Agg_edge_);
+    Agg_vertex_.Swap(graph_topology.Agg_vertex_);
+    face_Agg_.Swap(graph_topology.face_Agg_);
+    Agg_face_.Swap(graph_topology.Agg_face_);
+    face_edge_.Swap(graph_topology.face_edge_);
+
+    pAggExt_vertex_ = std::move(graph_topology.pAggExt_vertex_);
+    pAggExt_edge_ = std::move(graph_topology.pAggExt_edge_);
+
+    face_bdratt_.Swap(graph_topology.face_bdratt_);
+
+    Swap(vertex_start_, graph_topology.GetVertexStart());
+    Swap(edge_start_, graph_topology.GetEdgeStart());
+    Swap(aggregate_start_, graph_topology.GetAggregateStart());
+    Swap(face_start_, graph_topology.GetFaceStart());
+}
+
+std::vector<GraphTopology> MultilevelGraphTopology(
+    mfem::SparseMatrix& vertex_edge, const mfem::HypreParMatrix& edge_d_td,
+    const mfem::SparseMatrix* edge_boundaryattr, int num_levels, int coarsening_factor)
+{
+    std::vector<GraphTopology> graph_topologies;
+    graph_topologies.reserve(num_levels - 1);
+
+    // Construct finest level graph topology
+    mfem::Array<int> partitioning;
```

Comment:
Partitioning `vertex_vertex` is repeated a few times throughout the code/examples.  Maybe we should make a `PartitionAAT(vertex_edge, partitioning, coarsening_factor` utility function.  Not a huge deal tho, its only a few lines of code.